### PR TITLE
Add support Discover isPlayed/markPlayed/markUnplayed

### PR DESCRIPTION
**Pull request by glensc**
## Description

Fixes https://github.com/pkkid/python-plexapi/issues/1067

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated the docstring for new or existing methods
- [ ] I have added tests when applicable

### CODE REVIEW

**Comment by glensc** [comment]
need to add test to see if isWatched needs to be hacked too to fetch from myPlex

**Comment by glensc** [comment]
> need to add test to see if isWatched needs to be hacked too to fetch from myPlex. or rather throw to suggest to use myPlex directly for online media?

yes. media.isPlayed does not have online source metadata, so after marking played in cloud, the property is always False.

here's my test script

```python
from plexapi.server import PlexServer
from plexapi.video import Movie


def test_discover(env):
    url = env.get('PLEX_URL')
    token = env.get('PLEX_TOKEN')
    plex = PlexServer(baseurl=url, token=token)
    account = plex.myPlexAccount()
    # https://app.plex.tv/desktop/#!/provider/tv.plex.provider.discover/details?key=%2Flibrary%2Fmetadata%2F5d77696a7a53e9001e6e5da0
    result = account.searchDiscover("Olympus Has Fallen", libtype="movie")
    movie: Movie = result[0]

    print(movie)
    print("isPlayed: ", movie.isPlayed)
    print(movie.markPlayed())
    print("isPlayed: ", movie.isPlayed)


if __name__ == "__main__":
    import os
    import logging

    from plexapi import log as logger
    from plexapi import loghandler

    logger.removeHandler(loghandler)
    logger.setLevel(logging.DEBUG)
    logging.basicConfig(format="%(asctime)s:%(name)s: %(message)s", level=logging.DEBUG)

    test_discover(os.environ)

```

**Comment by glensc** [comment]
this is an ugly hack, I'm sure this should be solved differently, perhaps by adding a new Movie type, DiscoverMovie that has similar properties as Movie.

Or just leave as is and do the code in user code (outside PlexAPI)

**Comment by glensc** [comment]
here's one thing I noticed when testing, `userState.viewCount` gets incremented every time `markPlayed()` is called, but `markUnplayed()` resets play count to 0, not just decrementing

**Comment by glensc** [comment]
or should the API be similar to `MyPlexAccount.userState`?

```py
server = plex.myPlexAccount()
server.markPlayed(movie)
server.markUnplayed(movie)
```

**Comment by glensc** [comment]
ok, at 035c446, can use myplex. this seems more cleaner approach than hacking `PlayedUnplayedMixin`

```python
from plexapi.server import PlexServer
from plexapi.video import Movie


def test_discover(env):
    url = env.get('PLEX_URL')
    token = env.get('PLEX_TOKEN')
    plex = PlexServer(baseurl=url, token=token)
    account = plex.myPlexAccount()
    # https://app.plex.tv/desktop/#!/provider/tv.plex.provider.discover/details?key=%2Flibrary%2Fmetadata%2F5d77696a7a53e9001e6e5da0
    result = account.searchDiscover("Olympus Has Fallen", libtype="movie")
    movie: Movie = result[0]

    print(movie)
    print(movie.guid)
    print("isPlayed: ", account.isPlayed(movie))
    print(account.markPlayed(movie))
    print("isPlayed: ", account.isPlayed(movie))
    print(account.markUnplayed(movie))
    print("isPlayed: ", account.isPlayed(movie))


if __name__ == "__main__":
    import os
    import logging

    from plexapi import log as logger
    from plexapi import loghandler

    logger.removeHandler(loghandler)
    logger.setLevel(logging.DEBUG)
    logging.basicConfig(format="%(asctime)s:%(name)s: %(message)s", level=logging.DEBUG)

    test_discover(os.environ)
```

**Comment by glensc** [comment]
Maintainers: What version do I keep then? myPlex or PlayedUnplayedMixin?

I think myPlex is cleaner:
- https://github.com/pkkid/python-plexapi/pull/1090#issuecomment-1378971206

**Comment by JonnyWong16** [comment]
I agree. Use the myplex account as a temporary solution until proper support for Discover can be added.

**Comment by glensc** [comment]
Cleaned up

File: plexapi/myplex.py [hunk 1]
```diff
@@ -936,6 +936,27 @@ def userState(self, item):
         data = self.query(f"{self.METADATA}/library/metadata/{ratingKey}/userState")
         return self.findItem(data, cls=UserState)
 
+    def isPlayed(self, item):
+        """ Returns True if this item is played. """
```
Comment (JonnyWong16) [inline plexapi/myplex.py:940]: Add documentation for the `item` parameter.

Explicitly state that `item` must be a video object from `searchDiscover()`.

Same for the other two methods.

**Review by JonnyWong16 — CHANGES_REQUESTED** [review]

File: plexapi/myplex.py (hunk 1 above)
Comment (glensc) [reply plexapi/myplex.py:940]: item can be also video object if it's from new plex agent. i.e media object with guid `plex://<hash>`

Comment (JonnyWong16) [reply plexapi/myplex.py:940]: Right, I forgot about that. Then just the same as the `item` parameter from `userState()` above.

But update the doc sting to say "Return True if the item is played on Discover." (same for the other two methods) so it doesn't get confused with the mixin methods.

Comment (glensc) [reply plexapi/myplex.py]: Added: e986b56

agents doc:
- https://support.plex.tv/articles/200241558-agents/

Comment (glensc) [reply plexapi/myplex.py]: Added discover note: c61c733

**Review by JonnyWong16 — APPROVED** [review]

**Comment by RileyXX** [comment]
Great work guys! Thanks for the implementation.

**Pull request merged**